include/onnxruntime/core/graph/graph.h

```diff
@@ -283,8 +283,12 @@ class Node {
   void ToProto(ONNX_NAMESPACE::NodeProto& proto) const;
 
   /** Call the provided function for all explicit inputs, implicit inputs, and outputs of this Node.
-      If the NodeArg is an explicit or implicit input, is_input will be true when func is called. */
-  void ForEachDef(std::function<void(const onnxruntime::NodeArg&, bool is_input)> func) const;
+      If the NodeArg is an explicit or implicit input, is_input will be true when func is called. 
+	  @param include_missing_optional_defs Include NodeArgs that are optional and were not provided 
+	                                       i.e. NodeArg::Exists() == false.
+	  */
+  void ForEachDef(std::function<void(const onnxruntime::NodeArg&, bool is_input)> func,
+                  bool include_missing_optional_defs = false) const;
```

Comment:
>include_missing_optional_defs [](start = 23, length = 29)

How about OnlyIterateExistingDefs with default of True?

Reply:
exclude_missing_optional_defs?

---
In reply to: [252392502](https://github.com/Microsoft/onnxruntime/pull/410#discussion_r252392502) [](ancestors = 252392502)

Reply:
Was stumbled with the concept of implicit/explicit/optional input at the header, but after checking graph.cc I think the name is fine.

---
In reply to: [252437302](https://github.com/Microsoft/onnxruntime/pull/410#discussion_r252437302) [](ancestors = 252437302,252392502)